app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.html

```diff
@@ -18,16 +18,50 @@
               ng-model="$ctrl.command.payloadJSON"
               ng-change="$ctrl.updatePayload()"/>
 
-    <div class="form-group row slide-in" ng-if="$ctrl.command.invalid">
-      <div class="col-sm-9 col-sm-offset-3 error-message">
-        Error: {{$ctrl.command.errorMessage}}
+      <div class="form-group row slide-in" ng-if="$ctrl.command.invalid">
+        <div class="col-sm-9 col-sm-offset-3 error-message">
+          Error: {{$ctrl.command.errorMessage}}
+        </div>
       </div>
-    </div>
+  </stage-config-field>
+  <stage-config-field label="Custom Headers" help-key="pipeline.config.webhook.customHeaders">
+    <table class="table table-condensed packed">
+      <thead>
+        <tr>
+          <th style="width:40%">Key</th>
+          <th style="width:60%">Value</th>
+          <th class="text-right">Actions</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr ng-repeat="(key,value) in $ctrl.stage.customHeaders">
+          <td>
+            <strong class="small">{{key}}</strong>
+          </td>
+          <td>
+            <input type="text" required ng-model="$ctrl.stage.customHeaders[key]" value="{{value}}"
+                                                                                  class="form-control input-sm"/>
```

Comment:
same as below with the indentation; also, do you need the `value` attribute here? I would hope (but am not certain) that the `ng-model` attribute takes care of setting the value for you.